### PR TITLE
feat: pg bouncer for pg clusters

### PR DESCRIPTION
**Pull request by wai-wong-edb**

### CODE REVIEW

File: pkg/provider/resource_cluster.go [hunk 1]
```diff
@@ -403,6 +415,40 @@ func (c *clusterResource) Schema(ctx context.Context, req resource.SchemaRequest
...
+				MarkdownDescription: "Pg bouncer.",
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.Object{
+					plan_modifier.CustomPgBouncer(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"is_enabled": schema.BoolAttribute{
+						MarkdownDescription: "Is pg bouncer enabled.",
+						Required:            true,
```
Comment (edbamolvyavahare) [inline pkg/provider/resource_cluster.go:428]: should be optional as,  PgBouncer is not applicable on BAH clusters

Comment (wai-wong-edb) [reply pkg/provider/resource_cluster.go:428]: Maybe it has just released but I can turn on PgBouncer for BAH clusters using the UI on dev and prod. But PgBouncer here is optional, just the is_enabled field is required. There is a PgBouncer object above this field that is optional

File: examples/resources/biganimal_cluster/ha/resource.tf [hunk 2]
```diff
@@ -78,7 +78,23 @@ resource "biganimal_cluster" "ha_cluster" {
   read_only_connections = true
   region                = "us-east-1"
   superuser_access      = true
-  pgvector              = true
+  pgvector              = false
+
+  pg_bouncer = {
+    is_enabled = false
+    #  settings = [ # uncomment and set settings if is_enabled is true
```
Comment (edbamolvyavahare) [inline examples/resources/biganimal_cluster/ha/resource.tf:85]: If is_enabled is true, remove the comment and enter the settings. Should you prefer something different from the defaults.

File: examples/resources/biganimal_cluster/single_node/aws/resource.tf [hunk 3]
```diff
@@ -79,7 +79,23 @@ resource "biganimal_cluster" "single_node_cluster" {
   read_only_connections = false
   region                = "us-east-1"
   superuser_access      = true
-  pgvector              = true
+  pgvector              = false
+
+  pg_bouncer = {
+    is_enabled = false
+    #  settings = [ # uncomment and set settings if is_enabled is true
```
Comment (edbamolvyavahare) [inline examples/resources/biganimal_cluster/single_node/aws/resource.tf:86]: [same as above](https://github.com/EnterpriseDB/terraform-provider-biganimal/pull/421/files#r1442612360)

File: examples/resources/biganimal_cluster/single_node/azure/resource.tf [hunk 4]
```diff
@@ -79,7 +79,23 @@ resource "biganimal_cluster" "single_node_cluster" {
   read_only_connections = false
   region                = "eastus2"
   superuser_access      = true
-  pgvector              = true
+  pgvector              = false
+
+  pg_bouncer = {
+    is_enabled = false
+    #  settings = [ # uncomment and set settings if is_enabled is true
```
Comment (edbamolvyavahare) [inline examples/resources/biganimal_cluster/single_node/azure/resource.tf:86]: [same as above](https://github.com/EnterpriseDB/terraform-provider-biganimal/pull/421/files#r1442612360)

File: examples/resources/biganimal_cluster/single_node/bah_aws/resource.tf [hunk 5]
```diff
@@ -77,7 +77,23 @@ resource "biganimal_cluster" "single_node_cluster" {
   cloud_provider        = "bah:aws"
   read_only_connections = false
   region                = "us-east-1"
-  pgvector              = true
+  pgvector              = false
+
+  pg_bouncer = {
+    is_enabled = false
+    #  settings = [ # uncomment and set settings if is_enabled is true
```
Comment (edbamolvyavahare) [inline examples/resources/biganimal_cluster/single_node/bah_aws/resource.tf:84]: same as above

File: examples/resources/biganimal_cluster/single_node/bah_azure/resource.tf [hunk 6]
```diff
@@ -78,7 +78,24 @@ resource "biganimal_cluster" "single_node_cluster" {
   cloud_provider        = "bah:azure"
   read_only_connections = false
   region                = "eastus2"
-  pgvector              = true
+  pgvector              = false
+
+  pg_bouncer = {
+    is_enabled = false
+    #  settings = [ # uncomment and set settings if is_enabled is true
```
Comment (edbamolvyavahare) [inline examples/resources/biganimal_cluster/single_node/bah_azure/resource.tf:85]: same as above

File: examples/resources/biganimal_cluster/single_node/bah_gcp/resource.tf [hunk 7]
```diff
@@ -71,7 +71,23 @@ resource "biganimal_cluster" "single_node_cluster" {
   cloud_provider        = "bah:gcp"
   read_only_connections = false
   region                = "europe-west1"
-  pgvector              = true
+  pgvector              = false
+
+  pg_bouncer = {
+    is_enabled = false
+    #  settings = [ # uncomment and set settings if is_enabled is true
```
Comment (edbamolvyavahare) [inline examples/resources/biganimal_cluster/single_node/bah_gcp/resource.tf:78]: same as above, pl update at all applicable places

File: pkg/provider/data_source_cluster.go [hunk 8]
```diff
@@ -340,6 +340,37 @@ func (c *clusterDataSource) Schema(ctx context.Context, req datasource.SchemaReq
...
+				MarkdownDescription: "Pg bouncer.",
+				Optional:            true,
+				Computed:            true,
+				Attributes: map[string]schema.Attribute{
+					"is_enabled": schema.BoolAttribute{
+						MarkdownDescription: "Is pg bouncer enabled.",
+						Required:            true,
+					},
+					"settings": schema.SetNestedAttribute{
+						Description: "Pg bouncer settings.",
```
Comment (edbamolvyavahare) [inline pkg/provider/data_source_cluster.go:353]: PgBouncer Configuration Settings

File: pkg/provider/resource_cluster.go [hunk 9]
```diff
@@ -403,6 +415,43 @@ func (c *clusterResource) Schema(ctx context.Context, req resource.SchemaRequest
...
+					},
+					"settings": schema.SetNestedAttribute{
+						Description: "Pg bouncer settings.",
```
Comment (edbamolvyavahare) [inline pkg/provider/resource_cluster.go:431]: PgBouncer Configuration Settings

**Pull request merged**